casper/src/main/rholang/NonNegativeNumber.rho

```diff
@@ -0,0 +1,42 @@
+//scalapackage coop.rchain.casper.genesis.contracts
+
+//This is a contract to explicitly implement the int >= 0 behavioural type
+contract @"NonNegativeNumber"(@init, return) = {
+  new this, valueStore in {
+    contract @[*this, "add"](@x, success) = {
+      match x >= 0 {
+        true => {
+          for(@v <- valueStore){
+            valueStore!(v + x) | success!(true)
```

Comment:
At the moment, numbers are signed 32-bit integers.  You have to test for overflow.

Reply:
It would be nice to only provide arbitrary precision numbers in Rholang, in order to avoid overflows and provide a safer environment for the developers.

Reply:
@odeac That's the plan.